File: 가장_먼_노드.py
def tablize(n, edge):
    table = [[0 for _ in range(n)] for _ in range(n)]
    
    for i,j in edge:
        table[i-1][j-1] = 1
        table[j-1][i-1] = 1
        
    return table

def solution(n, edge):
    visited = [False for _ in range(n)]
    visited[0] = True
    
    graph = tablize(n, edge)
    bfs = [0]
    while True:
        next_bfs = []
        for i in bfs:
            connected = [j for j in range(n) if graph[i][j] == 1]
            for j in connected:
                if not visited[j]:
                    visited[j] = True
                    next_bfs.append(j)
        if not next_bfs:
            break
        bfs = next_bfs
    
    return len(bfs)

File: golang/Level 3/가장_먼_노드.go
func table(n int, edge [][]int) [][]bool {
    connected := make([][]bool, n)
    for i := range connected {
        connected[i] = make([]bool, n)
    }
    
    for _, points := range edge {
        connected[points[0]-1][points[1]-1] = true
        connected[points[1]-1][points[0]-1] = true
    }
    
    return connected
}

func solution(n int, edge [][]int) int {
    visited := make([]bool, n)
    visited[0] = true
    
    connected := table(n, edge)
    bfs := []int{0}
    for {
        next_bfs := []int{}
        for _, i := range bfs {
            for j, ok := range connected[i] {
                if ok && !visited[j] {
                    visited[j] = true
                    next_bfs = append(next_bfs, j)
                }
            }
        }
        if len(next_bfs) == 0 {
            break
        }
        bfs = next_bfs
    }
    
    return len(bfs)
}